datanode/dehistory/snapshot/service_load_snapshot.go

```diff
@@ -23,32 +26,58 @@ type Conn interface {
 	QueryRow(ctx context.Context, sql string, args ...interface{}) pgx.Row
 }
 
+type LoadResult struct {
+	LoadedFromHeight int64
+	LoadedToHeight   int64
+	RowsLoaded       int64
+}
+
 func (b *Service) LoadAllSnapshotData(ctx context.Context, currentStateSnapshot CurrentState,
 	contiguousHistory []History, sourceDir string,
-) (int64, error) {
+) (LoadResult, error) {
 	if err := killAllConnectionsToDatabase(ctx, b.connConfig); err != nil {
-		return 0, fmt.Errorf("failed to kill all connections to database: %w", err)
+		return LoadResult{}, fmt.Errorf("failed to kill all connections to database: %w", err)
 	}
 
 	vegaDbConn, err := pgxpool.Connect(context.Background(), b.connConfig.GetConnectionString())
 	if err != nil {
-		return 0, fmt.Errorf("unable to connect to vega database:%w", err)
+		return LoadResult{}, fmt.Errorf("unable to connect to vega database:%w", err)
+	}
+
+	datanodeBlockSpan, err := sqlstore.GetDatanodeBlockSpan(ctx, b.connConfig)
+	if err != nil {
+		return LoadResult{}, fmt.Errorf("failed to check if datanode has data:%w", err)
+	}
+
+	historyFromHeight := contiguousHistory[0].HeightFrom
+	historyToHeight := contiguousHistory[len(contiguousHistory)-1].HeightTo
+
+	if err = validateSpanOfHistoryToLoad(datanodeBlockSpan, historyFromHeight, historyToHeight); err != nil {
+		return LoadResult{}, fmt.Errorf("validation of history span to load failed: %w", err)
```

Comment:
I don't think we need to wrap the error here, we are wrapping far too much. The context given by the errors returned from `validateSpanOfHistory()` is more than enough.

Reply:
sorted